### ISSUE TITLE
Track name

### ISSUE THREAD

**Issue by wpysz** [Raruto/leaflet-elevation#72]
Here is your sample script: https://raruto.github.io/leaflet-elevation/examples/leaflet-elevation_almost-over.html

And here is the same script, but I added one waypoint to the via-emilia.gpx file:
https://ciekawe.tematy.net/testy/le/leaflet-elevation_almost-over.html

Then the checkbox for layersControl displays wpt->name, meaning "This is the name of the waypoint :-)" instead of trk->name "via-emilia.gpx"

![wpt-name](https://user-images.githubusercontent.com/70205437/91168814-8b23f200-e6d6-11ea-8f6d-1f9c5f9476b7.jpg)

**Comment by Raruto** [comment]
Hi wpsyz,
this happens due to the following line:

```js
controlElevation.on('eledata_loaded', function(e) {
  controlLayer.addOverlay(e.layer, e.name);
});
```

the name you see is added in this line of code: [leaflet-elevation/src/utils.js#L105](https://github.com/Raruto/leaflet-elevation/blob/464a5ed0aa98691e970d4c943b268897624864b9/src/utils.js#L105)

the `get_name()` function is native to the [leaflet-gpx](https://github.com/mpetazzoni/leaflet-gpx#available-functions) library, and this issue is related to the following line of code: [mpetazzoni/leaflet-gpx/gpx.js#L307](https://github.com/mpetazzoni/leaflet-gpx/blob/98772565059f832326bb4d7594b5c0bdfd0aa94d/gpx.js#L307)

in a nutshell, if you omit the `<name>` meta tag of your gpx file, the first `<name>` tag found is used (that is, the one related to the first waypoint).

To solve you can choose to:

- add the <name> meta tag yourself

  ```xml
  <xml>
  <gpx>
  <metadata>
      <name>via-emilia.gpx</name>
  </metadata>
  <trk>
    <wpt>
      <name>This is the name of the waypoint</name>
      ....
    </wpt>
  </trk>
  ```

- hardcode the name parameter

  ```js
  controlElevation.on('eledata_loaded', function(e) {
    controlLayer.addOverlay(e.layer, 'via-emilia.gpx');
  });
  ```

- propose a your own pull request

_Have a nice day,
Raruto_

**Comment by wpysz** [comment]
Thank you very much.
I will use one of the proposed methods. Although they are not perfect. 
In method one, I can't edit and save a GPX file with a standard track processing program (Oziexplorer, Mapsource, etc.) because they will remove <name> from the <metadata> section.
And the second way forces me to correct the code when changing the GPX file. 
But when I know the cause, I can handle it.
Will I find a manual somewhere describing the possibilities of using your tools to display maps and objects on maps? For now, I am learning how to use them by analyzing examples.

_Wojtek from Poland_

**Comment by Raruto** [comment]
> the second way forces me to correct the code when changing the GPX file.

If I understand correctly, such a thing may be enough for you:

```js
var gpx_url  = "https://raruto.github.io/leaflet-elevation/examples/via-emilia.gpx";
var gpx_name = gpx_url.substring(gpx_url.lastIndexOf('/')+1);
var elevation_options = { ... };

var controlElevation = L.control.elevation(elevation_options).addTo(map);
controlElevation.load(gpx_url);

controlElevation.on('eledata_loaded', function(e) {
  controlLayer.addOverlay(e.layer, gpx_name);
});
```

> Will I find a manual somewhere describing the possibilities of using your tools to display maps and objects on maps?

No just examples, but if you want to help someone else, this repository's [wiki](https://github.com/Raruto/leaflet-elevation/wiki) should be editable by anyone.

If it can help you:
- [here](https://github.com/Raruto/leaflet-elevation/blob/master/src/options.js) you can find all the default plugin options
- [here](https://github.com/Raruto/leaflet-elevation/search?q=fire&unscoped_q=fire) you can search all the available plugin hooks

_Happy coding,
Raruto_

**Comment by wpysz** [comment]
Thank you for good advice!
This third way of getting the name (from the GPX filename) will work best for me.
I have two last questions from the "newbie" series:
How do I assign my Google API key to the "pegman" functionality?
How do I move the control buttons (pegman, locate, zoom) from "bottomright" to "bottomleft"?

**Comment by Raruto** [comment]
> [How to enable/disable the leaflet user interface customizations?](https://github.com/Raruto/leaflet-elevation/wiki/FAQ#2-how-to-enabledisable-the-leaflet-user-interface-customizations)

Someone else has preceded you in adding this info in the wiki (although, my invitation to make it more detailed it is still valid...)

> How do I move the control buttons (pegman, locate, zoom) from "bottomright" to "bottomleft"?

Just pass it as a map option:

- [pegmanControl](https://github.com/Raruto/leaflet-ui/blob/master/src/leaflet-ui.js#L138)
- [zoomControl](https://github.com/Raruto/leaflet-ui/blob/master/src/leaflet-ui.js#L130)
- [locateControl](https://github.com/Raruto/leaflet-ui/blob/master/src/leaflet-ui.js#L142)
- etc.

> How do I assign my Google API key to the "pegman" functionality?

Take a look at the following option: [leaflet-pegman.js#L15](https://github.com/Raruto/leaflet-pegman/blob/master/leaflet-pegman.js#L15)

_Have nice day,
Raruto_
